### PR TITLE
Remove duplicated movedim implementation

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61940
* __->__ #61939

Differential Revision: [D29850798](https://our.internmc.facebook.com/intern/diff/D29850798)

### CODE REVIEW

File: aten/src/ATen/native/Batching.cpp [hunk 1]
```diff
@@ -70,27 +70,15 @@ static std::pair<Tensor,int64_t> remove_existing_batch_dim(
   return std::make_pair(std::move(result_tensor), newly_exposed_logical_dim);
 }
 
-// Poor man's version of np.moveaxis. Moves the dimension at `dst` to `src`
-// while preserving the order of other existing dimensions.
-// We should probably add np.moveaxis (it is more general) to PyTorch. (#36048)
-// When we do, replace the following with it.
-static Tensor movedim(const Tensor& self, int64_t src, int64_t dst) {
+// at::movedim but may return the original tensor if dst is the same as src.
```
Comment (peterbell10) [inline aten/src/ATen/native/Batching.cpp:73]: It turns out the behaviour here deviates slightly from `at::movedim` but I'd say it's at least worth highlighting that.

**Review by zou3519 — APPROVED** [review]
LGTM, can you rebase this PR? It looks like there are merge conflicts

**Comment by peterbell10** [comment]
@zou3519 rebased and CI is all passing now.

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29850798).

**Comment by zou3519** [comment]
Thank you! Attempting to land this now; if nothing happens in a couple of days please check in

**Pull request closed without merging**